cmake/developer_package/version.cmake

```diff
@@ -185,6 +185,33 @@ macro (addVersionDefines FILE)
     unset(__version_file)
 endmacro()
 
+macro (ov_add_version_defines FILE TARGET)
+    set(__version_file ${FILE})
+    if(NOT IS_ABSOLUTE ${__version_file})
+        set(__version_file "${CMAKE_CURRENT_SOURCE_DIR}/${__version_file}")
+    endif()
+    if(NOT EXISTS ${__version_file})
+        message(FATAL_ERROR "${FILE} does not exists in current source directory")
+    endif()
+    _remove_source_from_target(${TARGET} ${__version_file})
+    if (BUILD_SHARED_LIBS)
+        add_library(${TARGET}_version OBJECT ${__version_file})
+    else()
+        add_library(${TARGET}_version STATIC ${__version_file})
+    endif()
+    target_compile_definitions(${TARGET}_version PRIVATE
```

Comment:
looks like you also need to inherit:
- LTO settings
- [compilation options](https://cmake.org/cmake/help/latest/prop_sf/COMPILE_OPTIONS.html)